sqlobject/src/test/java/org/jdbi/v3/docs/TestInClauseExpansion.java

```diff
@@ -63,18 +65,21 @@ public void testInClauseExpansion() throws Exception
         ImmutableSet<String> findIdsForNames(@BindIn("names") List<Integer> names);
     }
 
-    public static class ImmutableSetCollectorFactory<T> implements CollectorFactory<T, ImmutableSet<T>> {
+    public static class ImmutableSetCollectorFactory<T> implements CollectorFactory {
 
         @Override
-        public boolean accepts(Class<?> type) {
-            return ImmutableSet.class.isAssignableFrom(type);
+        public boolean accepts(Type type) {
+            return getErasedType(type).equals(ImmutableSet.class);
         }
 
         @Override
-        public Collector<T, ImmutableSet.Builder<T>, ImmutableSet<T>> newCollector(Class<ImmutableSet<T>> type) {
-            return Collector.of(ImmutableSet.Builder::new, ImmutableSet.Builder::add, (first, second) -> {
-                throw new UnsupportedOperationException("Parallel collecting is not supported");
-            }, ImmutableSet.Builder::build, Collector.Characteristics.UNORDERED);
+        public Collector<T, ImmutableSet.Builder<T>, ImmutableSet<T>> newCollector(Type type) {
+            return Collector.of(
+                    ImmutableSet.Builder::new,
+                    ImmutableSet.Builder::add,
+                    (first, second) -> { throw new UnsupportedOperationException("Parallel collecting is not supported"); },
+                    ImmutableSet.Builder::build,
+                    Collector.Characteristics.UNORDERED);
```

Comment:
If this implementation is integral to the test, just call `GuavaCollectors.immutableSet()` rather than reproducing the whole thing.